README.md

```diff
@@ -3,13 +3,50 @@
+
+2. asymptotic complexity does not factor in external factors such as cashe misses, the temperature of the air, hardware accelarators and so on.
+our program's fisrt run at 1000 elements warmed up the CPU to an extent where it is underclocing to keep itself safe, this only happend because-
+you never clean your computer + your running a 350W threadripper with a stock cooler resting naked held securely by gravity ontop of your cpu. ![my computer before my real cooler arived](image-1.png)
+Now that the scene has been set, your seccond script, this time running a larger test of 10,000 elements, loads into memory. But tragicly it overflows into
+the swapfile! (as we only put 1 stick of 2 gb ram in our system...) The 50mb floppy disk we use as our prefered swapfile location is just enough to 
+pick up the slack and 100 secconds later after HARD but severely throttled number crunching (mostly disk reads) we calculate our final answer... 42!
+
+3. If we always ignore constant factors and mostly care about leading terms, then they can add up and lead to a significant discrepancy in our runtime.
+
 - Suppose finding a particular element in a binary search tree with 1,000
   elements takes 5 seconds. Given what you know about the asymptotic complexity
   of search in a binary search tree, how long would you guess finding the same
   element in a search tree with 10,000 elements takes? Explain your reasoning.
 
+i would say that the worst case would take:
+![6.66](calculator.png)
+6.6666 secconds
+
+reasoning: i set it up like ratio and solved for the missing variable. I assumed that-
+the scaling holdes and that there are no other variables affecting the runtime, ANd ig that it was the worst case i was calculating.
+I am assuming the complexity of the function to be the same as a properly balanced search tree where we cut the search space in half at each node.
+
 - You measure the time with 10,000 elements and it takes 100 seconds! List 3
   reasons why this could be the case, given that reasoning with the asymptotic
   complexity suggests a different time.
 
+1. Because it is a binary search tree, it is possible that it is not balanced. it is possible that it looks more like a linked list and that would
+make it more clear as to why it ran so slowly. Instead of chopping our search space in half each time in the worst case we could just be going down
+the linked list. in this worst case it would be a runtime of linear O(n) in the usual case it would be O(log(n)). Because here i am assuming the runtime is O(n) That would mean that to get a runtime of 100 seconds we were processing 100 elements a second, while the smaller input size was doing 200 elements a second. That difference in time and elements doesnt seam to line up to me, my conclusion here is that neither the smaller input or the larger input are balanced binary search trees but they arent the worst either, there somewhere in the middle.
```

Comment:
Not following your reasoning here. You assumed that for the smaller input the tree was balanced, but then in order to get both times it would have to be faster than when it's completely balanced if I understand you correctly. Please rephrase and explain in more detail.